lib/react/index.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import octicons from "octicons";
+
+export default class Octicon extends React.Component {
+  prepareAttributes(octicon) {
+    const { ariaLabel, medium, large } = this.props;
+    const attr = Object.assign({}, octicon.options);
+    // Default octicon sizes to small
+    attr["height"] = 16;
+
+    // Delete class set by octicons library
+    delete attr["class"];
+
+    // Read in small, medium, large props
+    if (medium) {
+      attr["height"] = 32;
+    } else if (large) {
+      attr["height"] = 64;
+    }
+
+    // Calculate the width based on height
+    attr["width"] =
+      attr["height"] * octicon.options["width"] / octicon.options["height"];
+
+    // If the user passed in aria-label
+    if (ariaLabel) {
+      attr["aria-label"] = ariaLabel;
+      attr["role"] = "img";
+
+      // Un-hide the icon
+      delete attr["aria-hidden"];
+    }
+
+    return attr;
+  }
+
+  render() {
+    const { name } = this.props;
+    const octicon = octicons[name];
+
+    const alignment = props => {
+      const { top, middle } = props;
+
+      if (top) {
+        return "text-top";
+      } else if (middle) {
+        return "middle";
+      }
+      return "text-bottom";
+    };
+
+    if (octicon) {
+      const svgStyle = {
+        display: "inline-block",
+        fill: "currentColor",
+        verticalAlign: alignment(this.props),
+        userSelect: "none"
+      };
+
+      return (
+        <svg
+          {...this.prepareAttributes(octicon)}
+          style={svgStyle}
+          /* eslint-disable-next-line react/no-danger */
+          dangerouslySetInnerHTML={{ __html: octicon.path }}
```

Comment:
It'd be ideal to avoid `dangerouslySetInnerHTML` for performance reasons. When this component is marked as *dirty*, react will have to update the DOM using `innerHTML` rather than changing just the attribute that has changed.

https://medium.com/@paularmstrong/twitter-lite-and-high-performance-react-progressive-web-apps-at-scale-d28a00e780a3#3732

Since all the SVGs are static, we should be able to head of time generate the `React.createElement` tree builder calls.